### PR TITLE
Rephrase C.21 to talk about declaring not defining

### LINKED ISSUE #870: Do =default destructors lead to violations of C.21 ?
**C.21 If you define or =delete any default operation, define or =delete them all**

states:

Reason:
The semantics of the special functions are closely related, so if one needs to be non-default, the odds are that others need modification too.
<https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#reason-76>

and:

Enforcement:
A class should have a declaration (even a =delete one) for either all or none of the special functions.
<https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#enforcement-71>

clang-tidy has a check, cppcoreguidelines-special-member-functions, <https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-special-member-functions.html>

which currently (by default) warns on

```
struct A
{
  virtual ~A() = default;
};

class 'A' defines a destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator [cppcoreguidelines-special-member-functions]
```

Is this correct behaviour? It follows the enforcement rule (which talks about declarations) but not the reason (which talks about non-default functions). 

There's an LLVM bug raised against this: https://bugs.llvm.org/show_bug.cgi?id=31858

I'm not trying to be pedantic here. Either behaviour is possible to implement but I want to ensure we're following the intent of C.21.

### PR DESCRIPTION
**Pull request by jwakely**
Add examples.

Fixes #870.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -4516,6 +4525,39 @@ This is known as "the rule of five" or "the rule of six", depending on whether y
 If you want a default implementation of a default operation (while defining another), write `=default` to show you're doing so intentionally for that function.
 If you don't want a default operation, suppress it with `=delete`.
 
+##### Example, good
+
+When a destructor needs to be declared just to make it `virtual`, it can be
+defined as defaulted. To avoid supressing the implicit move operations they
```
Comment (cubbimew) [inline CppCoreGuidelines.md]: Line 4531: supressing -> suppressing

Comment (jwakely) [reply CppCoreGuidelines.md]: Fixed, thanks

**Comment by johelegp** [comment]
Shouldn't it say non-default constructor special member functions, instead of special member functions? I think the default constructor doesn't participate in wether any other special member functions are implicitly generated or not.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -4516,6 +4525,39 @@ This is known as "the rule of five" or "the rule of six", depending on whether y
 If you want a default implementation of a default operation (while defining another), write `=default` to show you're doing so intentionally for that function.
 If you don't want a default operation, suppress it with `=delete`.
 
+##### Example, good
+
+When a destructor needs to be declared just to make it `virtual`, it can be
+defined as defaulted. To avoid suppressing the implicit move operations they
+they must also be declared, and then to avoid the class becoming move-only
```
Comment (MikeGitb) [inline CppCoreGuidelines.md]: here is an accidential doubling of "they"

Comment (jwakely) [reply CppCoreGuidelines.md]: Fixed, thanks.

**Comment by jwakely** [comment]
@johelegp I've pushed an update.

**Comment by AndrewPardoe** [comment]
Thank you!

**Pull request merged**